modules/new_serverpod_auth/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,589 @@
+import 'dart:isolate';
+import 'dart:typed_data';
+
+import 'package:image/image.dart';
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_profile_server/serverpod_auth_profile_server.dart';
+import 'package:serverpod_auth_profile_server/src/generated/protocol.dart';
+import 'package:serverpod_auth_profile_server/src/util/user_profile_extension.dart';
+import 'package:serverpod_shared/serverpod_shared.dart';
+
+/// Business logic for handling user profiles
+abstract final class UserProfiles {
+  /// Creates a new user and stores it in the database.
+  static Future<UserProfileModel> createUserProfile(
+    final Session session,
+    final UuidValue authUserId,
+    UserProfileData userProfile, {
+    final Transaction? transaction,
+  }) async {
+    return session.transactionOrSavepoint(
+      (final transaction) async {
+        final onBeforeUserProfileCreated =
+            UserProfileConfig.current.onBeforeUserProfileCreated;
+        if (onBeforeUserProfileCreated != null) {
+          userProfile = await onBeforeUserProfileCreated(
+            session,
+            authUserId,
+            userProfile,
+            transaction: transaction,
+          );
+        }
+
+        userProfile = userProfile.copyWith(
+          email: userProfile.email?.toLowerCase().trim(),
+        );
+
+        final createdProfile = await UserProfile.db.insertRow(
+          session,
+          UserProfile(
+            authUserId: authUserId,
+            userName: userProfile.userName,
+            fullName: userProfile.fullName,
+            email: userProfile.email,
+          ),
+        );
+
+        final createdProfileModel = createdProfile.toModel();
+
+        await UserProfileConfig.current.onAfterUserProfileCreated?.call(
+          session,
+          createdProfileModel,
+          transaction: transaction,
+        );
+
+        return createdProfileModel;
+      },
+      transaction: transaction,
+    );
+  }
+
+  /// Find a user profile by the `AuthUser`'s ID.
+  ///
+  /// By default the result is cached locally on the server. You can configure the cache
+  /// lifetime in [UserProfileConfig], or disable it on a call to call basis by
+  /// setting [useCache] to false.
+  ///
+  /// Throws a [UserProfileNotFoundException] in case no profile exists for the given [authUserId].
```

Comment:
```suggestion
  /// Throws a [UserProfileNotFoundException] in case no profile exists for the given [authUserId].
  ///
  /// In case a [transaction] is passed, the cache will not be used.
```